omegaconf/dictconfig.py

```diff
@@ -413,8 +413,8 @@ def get(self, key: DictKeyType, default_value: Any = DEFAULT_VALUE_MARKER) -> An
 
     def _get_impl(self, key: DictKeyType, default_value: Any) -> Any:
         try:
-            node = self._get_node(key=key)
-        except ConfigAttributeError:
+            node = self._get_node(key=key, throw_on_missing_key=True)
+        except (ConfigAttributeError, ConfigKeyError):
             if default_value != DEFAULT_VALUE_MARKER:
```

Comment:
Unrelated to this PR, but `is not` is a bit better than `!=` (avoids calling `__eq__()`)

Reply:
fixed